esmvaltool/diag_scripts/validation.py

```diff
@@ -22,11 +22,36 @@
     group_metadata,
     run_diagnostic,
 )
+from esmvaltool.diag_scripts.shared._base import ProvenanceLogger
 
 logger = logging.getLogger(os.path.basename(__file__))
 
 
-def plot_contour(cube, plt_title, file_name):
+def _get_provenance_record(cfg, plot_file, caption, loc):
+    """Create a provenance record describing the diagnostic data and plot."""
+    ancestor_files = [
+        k for k in cfg["input_data"].keys() if k.endswith(".nc")
```

Comment:
Are all input files used for all plots? This looks fishy.

Reply:
yes, they have identical names as them plots